Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -0,0 +1,438 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+#pragma once
+
+#include <optional>
+#include <variant>
+#include <SharedUtil.Template.h>
+#include "lua/CLuaOverloadParser.h"
+#include "lua/CLuaFunctionParseHelpers.h"
+#include "lua/CLuaStackChecker.h"
+#include "lua/LuaBasic.h"
+
+struct CLuaFunctionParserBase
+{
+    std::size_t iIndex = 1;
+    std::string strError = "";
+    std::string strErrorFoundType = "";
+
+    template <typename T>
+    inline void typeToNameVariant(SString& accumulator)
+    {
+        using param = typename is_variant<T>::param1_t;
+        if (accumulator.length() == 0)
+            accumulator = typeToName<param>();
+        else
+            accumulator += "/" + typeToName<param>();
+
+        if constexpr (is_variant<T>::count != 1)
+            return typeToNameVariant<is_variant<T>::rest_t>(accumulator);
+    }
+
+    template <typename T>
+    inline SString typeToName()
```

Comment:
Can we make this (and other methods) `constexpr`? (This compiles.)
```suggestion
    constexpr inline SString typeToName()
```

Reply:
Yes

Reply:
Is this actually necessary, or is the compiler smart enough to `constexpr` this method itself?

(it's obviously better to be explicit anyway, so later modifications don't mess it up.)

Reply:
Actually on second glance I don't think it's possible. SString (and std::string) are not constexpr.